azure-devops/run-build.yml

```diff
@@ -60,6 +60,14 @@ jobs:
         filename: 'azure-devops/validate-files.cmd'
         failOnStandardError: true
         arguments: '$(Build.ArtifactStagingDirectory)/tools/validate/validate.exe'
+    - task: PowerShell@2
+      displayName: 'Get Test Parallelsim'
+      timeoutInMinutes: 2
+      inputs:
+        targetType: inline
+        script: |
+          $number_of_parallel_tests = $env:NUMBER_OF_PROCESSORS - 2
+          Write-Output "##vso[task.setvariable variable=NUMBER_OF_PROCESSORS;]$number_of_parallel_tests"

```

Comment:
I am confused by the dataflow here. I'm familiar with the environment variable `NUMBER_OF_PROCESSORS`, which you appear to be reading, subtracting 2, and storing in a local variable `number_of_parallel_tests`. Then you appear to be setting a "task variable" named `NUMBER_OF_PROCESSORS` to the lower value here, and using it as a CMake argument below. It's the reuse of the name `NUMBER_OF_PROCESSORS` that I find confusing (because it has a lower value than the original environment variable). Could you use a distinct name?

Reply:
Yes that is confusing, I'll rename the ADO variable.